### PR TITLE
Dev 1111 contributor portal rp branded

### PR DESCRIPTION
**Pull request by gurjittoptal**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
[DEV-1111]

#### What's this PR do?
provides custom-styling based on default page for an RP

#### Why are we doing this? How does it help us?
to make contributor dashboard similar to styling used by RP

#### How should this be manually tested?
Check instructions at link below
https://news-revenue-hub.atlassian.net/browse/DEV-1111?focusedCommentId=44443

#### Have automated unit tests been added?
no

#### How should this change be communicated to end users?
n/a

#### Are there any smells or added technical debt to note?
no

#### Has this been documented? If so, where?
no

#### What are the relevant tickets?
**(Note: Please use syntax [JIRA-123] to link any relevant tickets)**
[dev-1111]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
n/a

#### Are there next steps? If so, what? Have tickets been opened for them?

[DEV-1111]: https://news-revenue-hub.atlassian.net/browse/DEV-1111?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by benjaminEwhite — CHANGES_REQUESTED** [review]
Left some inline comments. 

Also, think there should be a few tests added (could be via Cypress or Jest, not sure what makes most sense here).  There should be tests that show the happy path for applying custom RP branding on the related user facing routes in the SPA. And there should be parallel tests showing the happy path for applying default branding when custom not set up.

File: spa/src/components/DashboardRouter.js [hunk 1]
```diff
@@ -26,6 +29,11 @@ const ContributorDashboard = lazy(() =>
 const PageEditor = lazy(() => componentLoader(() => import('components/pageEditor/PageEditor')));
 
 function DashboardRouter() {
+  const isContributorApp = isContributorAppPath();
```
[the remark this answers is not in the document]
Comment (gurjittoptal) [reply spa/src/components/DashboardRouter.js:32]: makes sense. divided on applying this fix in this PR or handling via another ticker. will check with McH

File: spa/src/components/contributor/ContributorEntry.js [hunk 2]
```diff
@@ -14,11 +14,12 @@ import Input from 'elements/inputs/Input';
 // Analytics
 import { useConfigureAnalytics } from 'components/analytics';
 
-function ContributorEntry() {
+function ContributorEntry(props) {
```
[the remark this answers is not in the document]
Comment (gurjittoptal) [reply spa/src/components/contributor/ContributorEntry.js]: fair point.

File: spa/src/components/DashboardRouter.js [hunk 3]
```diff
@@ -26,6 +29,11 @@ const ContributorDashboard = lazy(() =>
...
+  console.log(isContributorApp);
+
+  if (isContributorApp) return <ContributorRouter />;
```
[the remark this answers is not in the document]
Comment (gurjittoptal) [reply spa/src/components/DashboardRouter.js:34]: agreed.

File: spa/src/components/contributor/ContributorEntry.js [hunk 4]
```diff
@@ -43,10 +44,15 @@ function ContributorEntry() {
     }
   };
 
+  let portalName = `RevEngine`;
+  if (page?.revenue_program) {
+    portalName = page?.revenue_program.name;
```
[the remark this answers is not in the document]
Comment (gurjittoptal) [reply spa/src/components/contributor/ContributorEntry.js]: done

**Review by benjaminEwhite — APPROVED** [review]

**Review by x110dc — APPROVED** [review]

**Pull request merged**